exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/WebServer.java

```diff
@@ -264,19 +265,26 @@ private ServerConnector createHttpsConnector() throws Exception {
 
     final SslContextFactory sslContextFactory = new SslContextFactory();
 
-    if (config.hasPath(ExecConstants.HTTP_KEYSTORE_PATH) &&
-        !Strings.isNullOrEmpty(config.getString(ExecConstants.HTTP_KEYSTORE_PATH))) {
-      logger.info("Using configured SSL settings for web server");
-      sslContextFactory.setKeyStorePath(config.getString(ExecConstants.HTTP_KEYSTORE_PATH));
-      sslContextFactory.setKeyStorePassword(config.getString(ExecConstants.HTTP_KEYSTORE_PASSWORD));
-
-      // TrustStore and TrustStore password are optional
-      if (config.hasPath(ExecConstants.HTTP_TRUSTSTORE_PATH)) {
-        sslContextFactory.setTrustStorePath(config.getString(ExecConstants.HTTP_TRUSTSTORE_PATH));
-        if (config.hasPath(ExecConstants.HTTP_TRUSTSTORE_PASSWORD)) {
-          sslContextFactory.setTrustStorePassword(config.getString(ExecConstants.HTTP_TRUSTSTORE_PASSWORD));
-        }
+    final boolean hasPath = config.hasPath(ExecConstants.HTTP_KEYSTORE_PATH);
+    final boolean hasPassword = config.hasPath(ExecConstants.HTTP_KEYSTORE_PASSWORD);
+
+    // Check if both keypath and password are present or not
+    if (hasPath && hasPassword) {
+      final String pathValue = config.getString(ExecConstants.HTTP_KEYSTORE_PATH);
+      final String passwordValue = config.getString(ExecConstants.HTTP_KEYSTORE_PASSWORD);
+
+      // checking if any one of them is null or empty
+      if (!Strings.isNullOrEmpty(pathValue) && !Strings.isNullOrEmpty(passwordValue)) {
+        sslContextFactory.setKeyStorePath(pathValue);
+        sslContextFactory.setKeyStorePassword(passwordValue);
+      }
+
+      // Throwing an exception if anyone of them is null or empty
+      else {
+        throw new DrillbitStartupException("keystorepath and/or keystorepassword can't be empty.");
```

Comment:
Please provide a bit of a clearer message. Also, provide the actual path name. This will be the critical message for folks that made mistake, the Drillbit won't start, and they have to figure out what's what.

Maybe,

```
"To enable web UI security, both " + ExecConstants.HTTP_KEYSTORE_PATH + " and " + ... + " are required.";
```

Also, this is just for the Web UI? In that case, maybe just disable the web UI, but don't fail the Drillbit.

```
"Web UI misconfiguration: web UI is disabled."
```